vcd/datasource_vcd_solution_add_on_instance.go

```diff
@@ -0,0 +1,68 @@
+package vcd
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func datasourceVcdSolutionAddonInstance() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: datasourceVcdSolutionAddonInstanceRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Solution Add-On Instance Name",
+			},
+			"add_on_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Parent Solution Add-On ID",
+			},
+			"input": {
+				Type:        schema.TypeMap,
+				Computed:    true,
+				Description: "Key value map of Solution Add-On Instance specified input fields",
+			},
+			"rde_state": {
+				Type:        schema.TypeString,
+				Description: "Parent RDE state",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func datasourceVcdSolutionAddonInstanceRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	addOnInstance, err := vcdClient.GetSolutionAddonInstanceByName(d.Get("name").(string))
+	if err != nil {
+		return diag.Errorf("error retrieving Solution Add-On Instance: %s", err)
+	}
+
+	dSet(d, "add_on_id", addOnInstance.SolutionAddOnInstance.Prototype)
+	dSet(d, "name", addOnInstance.SolutionAddOnInstance.Name)
+	dSet(d, "rde_state", addOnInstance.DefinedEntity.DefinedEntity.State)
```

Comment:
Any chance of nil pointer here?

Reply:
Indeed, I have switched to using `addOnInstance.DefinedEntity.State()` which I made for this purpose.
I don't expect `addOnInstance.SolutionAddOnInstance` to be nil when Add On Instance is retrieved, and the other fields are not pointers so hopefully they should be good.